client/fire-webclient/src/main/kotlin/uk/ac/ebi/fire/client/api/FireWebClient.kt

```diff
@@ -68,8 +69,8 @@ internal class FireWebClient(
         return client.putForObjectAsync<FireApiFile>("/objects/$fireOid/publish")
     }
 
-    override suspend fun unpublish(fireOid: String) {
-        client.deleteAsync("/objects/$fireOid/publish")
+    override suspend fun unpublish(fireOid: String): FireApiFile {
+        return client.deleteForObjectAsync<FireApiFile>("/objects/$fireOid/publish")
```

Comment:
I think given api is already native async we do not need any more extensions to faciliated. I think we can simple do now 
```
   override suspend fun unpublish(fireOid: String): FireApiFile {
        return client.delete().uri("/objects/$fireOid/publish").retrieve().awaitBody()
    }
```

I mean extensions were good during transition period from  rest template to webclient but now IMO we can start getting familiar with real API